### ISSUE TITLE
Enable AKS Control Plane Logging during AKS cluster provisioning

### ISSUE THREAD

**Issue by jayush** [Azure/AKS#1443]
**What happened**:
At present we have to first provision an AKS cluster and then via Azure Diagnostic Settings enable logging for AKS control plane.

**What you expected to happen**:
Ideally we should be able to enable AKS control plane logging with cluster creation request itself.

**How to reproduce it (as minimally and precisely as possible)**:
Refer az aks create --help and Azure Go SDK. There is no option to enable control plane logging at cluster creation time.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Size of cluster (how many worker nodes are in the cluster?)
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.)
- Others:

**Comment by palma21** [comment]
You can enable via ARM. We can take this as a CLI feature request.

**Comment by blueelvis** [comment]
Is there any update on this?

**Comment by josh-ferrell** [comment]
@palma21 Bumping this, I don't see how this functionality exposed via CLI or even terraform. The instructions are limited to implementation via the web portal only. https://docs.microsoft.com/en-us/azure/aks/view-master-logs

For those of us limited to using a service principal for cluster creation and administration this greatly hinders management as well as automation efforts.

**Comment by shackit** [comment]
Hitting this problem too it seems impossible to turn on this pretty obvious enterprise feature/requirement. We are also using ARM for our resource deployment needs. I have been trying to reach out to our TAM, I have raised a support case but still it is unclear how to achieve this. 

The closest thing that I have seen is to use a Azure Policy Assignment - but this raises another issue that the roleDefinitions defined within the policy do not have the appropriate permission to actually apply the settings and make the AKS cluster compliant.

@palma21 if you have some insights into how this feature can be enabled by ARM I would really appreciate a link to a reference implementation or documentation to describe that process.

**Comment by josh-ferrell** [comment]
Just to update this, we were able to use terraform to create the resource "azurerm_monitor_diagnostic_setting".

Since TF is handling all the heavy lifting we are ok here although we have noticed were able to delete the AKS clusters and resource groups but the diagnostic settings remain even though they aren't exposed via the API or Portal. I'm not sure how that it but we see it if we try to create identically named resource groups & clusters a second time TF will error out saying the diag setting exist already.
